src/Samples/Toolkit.SampleApp.WPF/Samples/Downloading/DownloadItemVM.cs

```diff
@@ -0,0 +1,199 @@
+        {
+            if (oldValue != null)
+            {
+                oldValue.StatusChanged -= OnStatusChanged;
+                if (System.IO.File.Exists(JsonFilename))
+                    System.IO.File.Delete(JsonFilename);
+            }
+            if (newValue != null)
+            {
+                newValue.Progress += (s, e) => { _uiDispatcher(() => { OnPropertyChanged(nameof(Progress)); OnPropertyChanged(nameof(DownloadSpeed)); }); };
+                newValue.Completed += (s, e) =>
+                {
+                    _uiDispatcher(() =>
+                    {
+                        DownloadTask = null;
+                    });
+                };
+                newValue.StatusChanged += OnStatusChanged;
+                System.IO.File.WriteAllText(JsonFilename, DownloadTask.ToJson()); // Store json for later resuming 
+            }
+            OnStatusChanged(null, EventArgs.Empty);
+        }
+
+        [RelayCommand(CanExecute = nameof(CanDelete))]
+        private void DeleteDownload()
+        {
+            if (System.IO.File.Exists(Filename))
+            {
+                System.IO.File.Delete(Filename);
+                OnStatusChanged(null, EventArgs.Empty);
+            }
+        }
+        private bool CanDelete => Item is not null && System.IO.File.Exists(Filename);
+
+        public bool HasError => DownloadTask?.Exception is not null;
+
+        public double Progress
+        {
+            get
+            {
+                if (Item is null) return 0;
+                if (System.IO.File.Exists(Filename)) return 1; // Already downloaded
+                if (DownloadTask is null) return 0;
+                if (!DownloadTask.TotalLength.HasValue) return 0; // Unknown size
+                return DownloadTask.BytesDownloaded / (double)DownloadTask.TotalLength.Value;
+            }
+        }
+
+        public string DownloadSpeed
+        {
+            get
+            {
+                if (DownloadTask is null || DownloadTask.DownloadSpeed <= 0) return string.Empty;
+                if (DownloadTask.DownloadSpeed < 1048)
+                    return $"{DownloadTask.DownloadSpeed} bytes/s";
+                if (DownloadTask.DownloadSpeed < 1024 * 1024)
+                    return $"{DownloadTask.DownloadSpeed / 1024d:0.0} kb/s";
+                return $"{DownloadTask.DownloadSpeed / 1024d / 1024:0.0} mb/s";
+            }
+        }
+
+        public string Status
+        {
+            get
+            {
+                if (Item is null)
+                    return "Loading item data...";
+                if (DownloadTask is null)
+                    return System.IO.File.Exists(Filename) ? "Completed" : "Not started";
+                return DownloadTask.Status.ToString();
+            }
+        }
+
+        private string Filename => Item is null ? null : System.IO.Path.Combine(_folder, Item?.Name!); // Where we will save the file
```

Comment:
I wouldn't trust item Name to be a valid filename.  I would add something that [strips invalid characters out](https://stackoverflow.com/a/12800424).

Actually, I'm not sure if I could trust the Name to be unique either.  Would be it safer to just use an item ID for filename or possibly some combination of ID + sanitized Name?